### ISSUE TITLE
MA0007 shouldn't trigger on spread element over Linq operation when last value is followed by a comma

### ISSUE THREAD

**Issue by robv8r** [meziantou/Meziantou.Analyzer#1126]
### Version of the Meziantou.Analyzer NuGet package

3.0.63

### Rule Identifier

MA0007

### Target Framework

net10.0

### C# Language version

C# 14.0

### Description

The Analyzer raises MA0007 when using a spread element derived from a Linq operation even though the last item is followed by a comma.

### Reproduction Steps

Minimal code:

```c#
[ApiController]
[Route("[controller]")]
public class WeatherForecastsController : ControllerBase
{
    private static readonly string[] Summaries =
    [
        "Freezing", "Bracing", "Chilly", "Cool", "Mild", "Warm", "Balmy", "Hot", "Sweltering", "Scorching",
    ];

    [HttpGet(Name = "GetWeatherForecasts")]
    public IEnumerable<WeatherForecast> Get()
    {
        // MA0007 - Add comma after the last value
        return [.. Enumerable.Range(1, 5).Select(index => new WeatherForecast
        {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
        })];
    }
}
```

### Other information

The above code is derived from the standard ASP.NET Core Web API template.  

The original code looks similar to this:

```c#
return Enumerable.Range(1, 5).Select(index => new WeatherForecast
{
    Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
    TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
    Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
}).ToList();
```

The `ToList()` method raises [IDE0305](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/style-rules/ide0305).  Using the fix for IDE0305 results in the above "Minimal code".

**Comment by meziantou** [comment]
I think you are interested in MA0007, not MA0051 (Method is too long).
Indeed, MA0007 should be updated for the collection expression. I think a good tradeoff is to disable comma after the last value in a collection expression when the last value is followed directly with the closing bracket (`]`). If there is a new line between the last value and the closing bracket, then a comma should be required.

```
// ok
[.. Enumerable.Range(1, 5).Select(index => new WeatherForecast
        {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
        })]
```

```
// ok
[value, .. Enumerable.Range(1, 5).Select(index => new WeatherForecast
        {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
        })]
```

```
// ok
[
    value,
    .. Enumerable.Range(1, 5).Select(index => new WeatherForecast
       {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
       }),
]
```

```
// nok, should have a comma after the last value
[
    .. Enumerable.Range(1, 5).Select(index => new WeatherForecast
       {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
       })
]
```

```
// nok, should have a comma after the last value
[
    value,
    .. Enumerable.Range(1, 5).Select(index => new WeatherForecast
       {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
       })
]
```

```
// nok, should have a comma after the last value
[
    value,
    .. Enumerable.Range(1, 5).Select(index => new WeatherForecast
       {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
       }) // comment
]
```

```
// ok, should have a comma after the last value
[
    value,
    .. Enumerable.Range(1, 5).Select(index => new WeatherForecast
       {
            Date = DateOnly.FromDateTime(DateTime.Now.AddDays(index)),
            TemperatureC = RandomNumberGenerator.GetInt32(-20, 55),
            Summary = Summaries[RandomNumberGenerator.GetInt32(Summaries.Length)],
       })]
```

**Comment by robv8r** [comment]
Thank you for the feedback, you are correct.  I've updated the issue to reflect `MA0007`.
